kong/resty/dns_client/README.md

```diff
@@ -0,0 +1,117 @@
+Name
+====
+
+The module is currently Kong only, and builds on top of the `lua-resty-dns` and the kong's `lua-resty-mlcache` library.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [APIs](#apis)
+    * [new](#new)
+    * [resolve](#resolve)
+
+# APIs
+
+The following APIs are for internal development use only within Kong. In the current version, the new DNS library still needs to be compatible with the original DNS library. Therefore, the functions listed below cannot be directly invoked. For example, the `_M:resolve` function in the following APIs will be replaced to ensure compatibility with the previous DNS library API interface specifications `_M.resolve`.
+
+## new
+
+**syntax:** *c, err = dns_client.new(opts)*  
+**context:** any
+
+**Functionality:**
+
+Creates a dns client object. Returns nil and a message string on error.
+
+Perform a series of initialization operations:
+
+* parse `host` file
+* parse `resolv.conf` file (used by the underlying `lua-resty-dns` library)
+* initialize multiple TTL options
+* create a mlcache object and initialize it
+
+**Input paramenters:**
+
+`@opts` It accepts a options table argument. The following options are supported:
+
+* TTL options
+  * `valid_ttl`
+    * same to the option `dns_valid_ttl` in kong.conf
+  * `stale_ttl`
+    * same to the option `dns_stale_ttl` in kong.conf
+  * `empty_ttl`
+    * same to the option `dns_not_found_ttl` in kong.conf
+  * `bad_ttl`
+    * same to the option `dns_error_ttl` in kong.conf
+* `hosts` (default: `/etc/hosts`)
+  * the path of `hosts` file
+* `resolv_conf` (default: `/etc/resolv.conf`)
+  * the path of `resolv.conf` file, it will be parsed and passed into the underlying `lua-resty-dns` library.
```

Comment:
Cool, first to know, seems to support them, we still have a long way to go.

I agree we could support them in the further plan, currently this PR only re-implemented hosts file parsing logic. It didn't reimplement `resolv` file parsing logic, which is so complicated, including `utils.applyEnv` function and `utils.parseResolvConf` function.

So I left a `TODO` there

```
-- TODO: need to rewrite it instead of calling parseResolvConf
function _M.parse_resolv_conf(path, enable_ipv6)
```